### DISCUSSION TITLE
[psql] Is there a built-in way to use `union`?

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#3780] [Q&A] [answered]
I’d like to use [`union`](https://www.postgresql.org/docs/current/queries-union.html) in MikroORM. Is there a built-in way (like `qb.union()` or do I need to use `qb.raw()` for that?

For example, I want to get some rows from a table based on some condition, however, at any case, return at least two rows with the greatest ID.

```sql
(
  select * from sch.tbl
  where id > 10
  order by id desc
) union (
  select * from sch.tbl order by id desc limit 2
) order by id asc;
```

I can use Knex to accomplish this:

```ts
const knex = this.orm.em.getKnex()

const query = knex
  .withSchema('sch')
  .select('*')
  .from('tbl')
  .where('id', '>', 10)
  .orderBy('start_time', 'desc')
  .limit(10)
  .union(
    qb => qb
      .select('*')
      .withSchema('sch')
      .from('tbl')
      .orderBy('id', 'desc')
      .limit(2),
    true
  )
  .orderBy('start_time', 'asc')

const res = (await this.orm.em.getConnection().execute(query)).map(e => this.orm.em.map(StatusInterval, e))
```

Now, is there a MikroORM way of doing this (a syntactic sugar, like a `union()` method) in MikroORM? I didn’t find anything in the docs.

**Comment by B4nan** [comment, marked as answer]
Not really, but you can use the ORM QB, configure it the way you want, and convert it to configured knex QB via `qb.getKnexQuery()`, and combine those.

Adding syntax sugar around that could be quite simple, PR welcome.

**Reply by tukusejssirs** [reply]
Thanks for reply! :pray:

Just to note: I tried to use `qb.getKnexQuery()`, however, it always prepended `select *` to the SQL command in the OP (I think this is caused by Knex).

The only way I could create the query is like I posted in the OP. At first, I was surprised that I can use multiple `orderBy()` methods on `knex`, however Knex seems to understand it the way I expected it.
